usermods/Battery/usermod_v2_Battery.h

```diff
@@ -0,0 +1,783 @@
+     *     - Red color indicates a conflict.  Yellow color indicates a pin with a warning (e.g. an input-only pin)
+     *   - Tip: use int8_t to store the pin value in the Usermod, so a -1 value (pin not set) can be used
+     *
+     * See usermod_v2_auto_save.h for an example that saves Flash space by reusing ArduinoJson key name strings
+     * 
+     * If you need a dedicated settings page with custom layout for your Usermod, that takes a lot more work.  
+     * You will have to add the setting to the HTML, xml.cpp and set.cpp manually.
+     * See the WLED Soundreactive fork (code and wiki) for reference.  https://github.com/atuline/WLED
+     * 
+     * I highly recommend checking out the basics of ArduinoJson serialization and deserialization in order to use custom settings!
+     */
+    void addToConfig(JsonObject& root)
+    {
+      // created JSON object: 
+      /*
+      {
+        "Battery": {
+          "pin": "A0",                    <--- only when using esp32 boards
+          "calculate-time-left": 3600,    <-- time in seconds
+          "min-battery-voltage": 2.6, 
+          "max-battery-voltage": 4.2,
+          "total-battery-capacity": 3000, <-- in mAh
+          "calibration": 0,
+          "read-interval-ms": 30000,      <-- time in milliseconds
+          "auto-off": {
+            "enabled": true,
+            "threshold": 10,     <-- in percent
+          },
+          "low-power-indicator": {
+            "enabled": true,
+            "preset": 0,        <-- preset ID
+            "threshold": 10,    <-- in percent
+            "duration": 5,      <-- time in seconds
+          }  
+        }
+      }
+      */ 
+      JsonObject battery = root.createNestedObject(FPSTR(_name));           // usermodname
+      #ifdef ARDUINO_ARCH_ESP32
+        battery["pin"] = batteryPin;
+      #endif
+
+      // battery[F("calculate-time-left")] = calculateTimeLeftEnabled;
+      battery[F("min-battery-voltage")] = minBatteryVoltage;
+      battery[F("max-battery-voltage")] = maxBatteryVoltage;
+      battery[F("total-battery-capacity")] = totalBatteryCapacity;
+      battery[F("calibration")] = calibration;
+      battery[FPSTR(_readInterval)] = readingInterval;
+      
+      JsonObject ao = battery.createNestedObject(F("auto-off"));               // auto off section
+      ao[FPSTR(_enabled)] = autoOffEnabled;
+      ao[FPSTR(_threshold)] = autoOffThreshold;
+
+      JsonObject lp = battery.createNestedObject(F("low-power-indicator"));    // low power section
+      lp[FPSTR(_enabled)] = lowPowerIndicatorEnabled;
+      lp[FPSTR(_preset)] = lowPowerIndicatorPreset; // dropdown trickery (String)lowPowerIndicatorPreset; 
+      lp[FPSTR(_threshold)] = lowPowerIndicatorThreshold;
+      lp[FPSTR(_duration)] = lowPowerIndicatorDuration;
+
+      DEBUG_PRINTLN(F("Battery config saved."));
+    }
+
+    void appendConfigData()
+    {
+      oappend(SET_F("addInfo('Battery:min-battery-voltage', 1, 'v');"));
```

Comment:
Try to use shorter names for config variables as `oappend()` buffer size is only 3kB and may be too small if more than 1 usermod is enabled.

Reply:
Done. 👍
(see next commit)